cmd/cli/plugin/cluster/upgrade.go

```diff
@@ -154,6 +162,14 @@
 		return "", err
 	}
 
+	// TODO: update this condition after CLI fully support the package based LCM.
+	// Since CLI should support the pre package-based-lcm where the updatesAvailable condition was part of
+	// TKRs, code checking the TKRs for available upgrade should remain.
+	cluster, err := getClusterResource(clusterClient, clusterName, uc.namespace)
+	if err == nil && (cluster.Spec.Topology != nil || clusterHasUpdatesAvailableCondition(cluster)) {
```

Comment:
As discussed, I will update the condition check to only check for `clusterHasUpdatesAvailableCondition(cluster)` so that it is good for ClusterClass based clusters, legacy clusters and also to cater to the usecase where users ClusterClass but doesn't use TKR resolution/TKRs.